WordPress/openverse
# Add GitHub username link to changelog, unify & simplify format

## Fixes



Fixes #1974 by @sarayourfriend
(among a few other changes)

## Description



This PR makes a few changes to the changelog format and release draft rendering to unify the various templates, simplify the rendering, and add more context:

- PR authors are now linked to their GitHub profile in the sphinx rendered view
- Removed the "Credits" section, which was inaccurate (see [this comment](https://github.com/WordPress/openverse/pull/1980#discussion_r1183032150) for more info) and made redundant by both the individual `@` references to PR authors and GitHub's own "Contributors" section ([example](https://github.com/WordPress/openverse/releases/tag/catalog-2023.04.27.02.43.13))
- Changed the link format for pull requests to have the link only consist of the actual number, rather than the full PR title. This is consistent with some other changelogs I've seen ([example](https://airflow.apache.org/docs/apache-airflow/stable/release_notes.html#new-features)), along with how GitHub renders the links in release format too ([example](https://github.com/WordPress/openverse/releases/tag/catalog-2023.04.27.02.43.13)). It also makes the lines a bit easier to read since the entire line isn't the bright blue of a hyperlink (see below for an example).
- ~~Removed the `link_issues` extension - this was providing a tooltip for the link but that info was the exact same as the PR title which is supplied immediately before it, which makes that redundant. This also prevents having to use any GitHub API requests (described more in #1974) since we're no longer using GitHub to fill some of that information out.~~ No longer correct, see discussion below
- Retroactively applies all of the above to existing changelog files.

There might be more that the `link_issues` extension is doing that I'm unaware of, but we were already no longer leveraging it with the changes in #1973 so I think we're safe to remove it (which is one less thing to vendor and maintain!).

**Before**
![image](https://user-images.githubusercontent.com/10214785/236334914-a674c089-dc2f-48c8-8746-552753a78748.png)


**After**
![image](https://user-images.githubusercontent.com/10214785/236334985-9765f723-f256-4f6e-bf4e-2ba4e007d08b.png)


## Testing Instructions


1. Run `just documentation/live`
1. Go through each existing changelog (or spot check) and ensure that usernames link to GitHub profiles, and PR numbers link to the appropriate PRs.

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
